### PR TITLE
MaxText shardings.py test

### PR DESCRIPTION
**Pull request by khatwanimohit**
# Description

Multipod test for [pedagogical_examples/shardings.py](https://github.com/google/maxtext/blob/main/pedagogical_examples/shardings.py)

# Tests

Ran it locally: screen/3ZYPPzrToSRBDdw

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run one-shot tests and provided workload links above if applicable. 
- [X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: dags/multipod/legacy.py
Comment (jonb377) [inline dags/multipod/legacy.py:152]: The test will format in the accelerator and slice count, so this can just be `maxtext-shardings-{SetupMode.STABLE.value}`

File: dags/multipod/legacy.py
Comment (jonb377) [inline dags/multipod/legacy.py:157]: Do we only want to run these on stable?

**Review by jonb377 — COMMENTED** [review]
Thanks Mohit!

File: dags/multipod/legacy.py
Comment (khatwanimohit) [reply dags/multipod/legacy.py:157]: In XLML we only run this on stable. But I think its makes sense to run on nightly as well. Will change that

File: dags/multipod/legacy.py
Comment (khatwanimohit) [reply dags/multipod/legacy.py:152]: Makes sense, should I also make the change to other test names `maxtext-perf` and `maxtext-checkpoint-reshard` in this PR?

Comment (jonb377) [reply dags/multipod/legacy.py:152]: Oh haha I didn't catch that in the other PRs, please do!

**Comment by jonb377** [comment]
@khatwanimohit Can we merge this?

**Comment by khatwanimohit** [comment]
> @khatwanimohit Can we merge this?

Not sure why its asking to get approval from other code owners. I thought you were the code owner as well

**Review by will-cromar — APPROVED** [review]

**Pull request merged**
